### ISSUE TITLE
pmin/pmax methods?

### ISSUE THREAD

**Issue by cdeterman** [viennacl/viennacl-dev#174]
It may be useful to have `pmin` and `pmax` functions whereby the `min` and `max` is applied over a `viennacl::vector` or `viennacl::matrix` object.  

For example

```
vec = [1,2,3,4,5]
pmax(vec, 3)
[3,3,3,4,5]

pmin(vec, 2)
[1,2,2,2,2]
```

This can be done with for loops but they would be grossly inefficient as sizes increase.

**Comment by karlrupp** [comment]
This is similar to https://github.com/viennacl/viennacl-dev/issues/155, isn't it? For example:

    result = element_max(vec, 3);

**Comment by cdeterman** [comment]
@karlrupp yes, that would be very similar but since I didn't see any `element_max` function in the documentation previously I thought it best to start another issue.
